src/RF24Log.h

```diff
@@ -0,0 +1,274 @@
+    }
+
+    /**
+     * @brief Log a message
+     * @param lvl The logging level used for the specified message.
+     * @param origin The orgin of the message.
+     * @param msg The specified message.
+     */
+    template <typename St, typename... Ts>
+    void logOrigin(uint8_t lvl, St origin, Ts... msg)
+    {
+        if ((lvl < level && level) || handler == nullptr)
+            return;
+
+        getLogger()
+#ifdef ARDUINO
+        << millis()
+#else // !defined(ARDUINO)
+        << timestamp()
+#endif // !defined(ARDUINO)
+        << ':';
+
+        if (lvl % 10 == 0)
+            getLogger() << (char*)pgm_read_ptr(&levelDesc[lvl / 10 - 1]) << ':';
+        else
+            getLogger() << "Lvl " << lvl << ':';
+
+        getLogger() << origin << ": ";
+        outputData(msg...) << endl;
+    }
+
+    /**
+     * @brief output a @ref INFO level message
+     * @param msg The message to output.
+     */
+    template <typename... Ts>
+    void info(Ts... msg)
+    {
+        log(INFO, msg...);
+    }
+
+    /**
+     * @brief output a @ref DEBUG level message
+     * @param msg The message to output.
+     */
+    template <typename... Ts>
+    void debug(Ts... msg)
+    {
+        log(DEBUG, msg...);
+    }
+
+    /**
+     * @brief output a @ref WARN level message
+     * @param msg The message to output.
+     */
+    template <typename... Ts>
+    void warn(Ts... msg)
+    {
+        log(WARN, msg...);
+    }
+
+    /**
+     * @brief output a @ref ERROR level message
+     * @param msg The message to output.
+     */
+    template <typename... Ts>
+    void error(Ts... msg)
+    {
+        log(ERROR, msg...);
+    }
+
+protected:
+
+#ifndef ARDUINO
+    /**
+     * @brief output a timestamp for the proceeding message
+     * @note this is for non-Arduino platforms
+     * @returns A c-string of the current time
+     */
+    char* timestamp()
+    {
+        time_t now;
+        time(&now);
+        return ctime(now);
+    }
+#endif
+
+    /**
+     * @brief output an arbitrary type of data
+     *
+     * This is our base case for parameter unpacking
+     * @param data The data to output
+     * @returns The instantiated output stream object passed to setHandler()
+     */
+    template <typename Tdata>
+    StreamType &outputData(Tdata data)
+    {
+        getLogger() << data;
+        return getLogger();
+    }
+
+    /**
+     * @brief output an arbitrary type of data
+     *
+     * This is our recursive scenario for parameter unpacking
+     * @param data The data to output
+     * @param rest the pack of sequential parameters
+     * @returns The instantiated output stream object passed to setHandler()
+     */
+    template <typename Tdata, typename... Rest>
+    StreamType &outputData(Tdata data, Rest... rest)
+    {
+        getLogger() << data;
+        return outputData(rest...);
+    }
+
+};
+
+/** @brief A convenient variable to get started quickly on supported platforms */
+RF24Logger<Stream_t> logging;
```

Comment:
This quite ok. Just the variable name should be some different, like RF24Logger or similar (I'm not quite sure if it is possible to name the variable the same as the class).

Reply:
Variable names (compared to class names) have a different naming convention in C++. To self-identify as an `RF24Logger` singleton, its name would be `rf24logger`.

Reply:
Ok, thanks.